From 1721a3df8542534c46b578c9d35dbe9b33f5ab5c Mon Sep 17 00:00:00 2001
From: Dmitry Ornatsky <cdo021c@motorola.com>
Date: Tue, 1 Sep 2015 07:50:23 -0500
Subject: [PATCH 059/959] IKKRNBSP-3351 arm/dt: 8952: add spi_4 interface for
 SH

Configure BLSP for spi_4 interface on athene.
Delete i2c_5 which uses the same BLSP.
Delete unused and disabled spi0 interface.

Change-Id: I87eae081b888d938411687f70ac8da503267d4d9
Signed-off-by: Dmitry Ornatsky <cdo021c@motorola.com>
Reviewed-on: http://gerrit.mot.com/780498
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi | 85 +++++++++++++++++++++++++
 1 file changed, 85 insertions(+)

diff --git a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
index 09630b68cef..d1672965259 100644
--- a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
@@ -17,6 +17,9 @@
 / {
 	aliases {
 		serial0 = &blsp1_uart2;
+		spi4 = &spi_4;
+		/delete-property/ i2c5;
+		/delete-property/ spi0;
 	};
 };
 
@@ -57,9 +60,91 @@
 			bias-pull-up; /* pull-up */
 		};
 	};
+
+	spi4_active {
+		/* BLSP5 MOSI, MISO, CLK */
+		qcom,pins = <&gp 16>, <&gp 17>, <&gp 19>;
+		qcom,num-grp-pins = <3>;
+		qcom,pin-func = <1>;
+		label = "spi4-active";
+		/* active state */
+		spi4_default: spi4_default {
+			drive-strength = <12>; /* 12 MA */
+			bias-disable = <0>; /* No PULL */
+		};
+	};
+
+	spi4_suspend {
+		/* BLSP5 MOSI, MISO, CLK */
+		qcom,pins = <&gp 16>, <&gp 17>, <&gp 19>;
+		qcom,num-grp-pins = <3>;
+		qcom,pin-func = <0>;
+		label = "spi4-suspend";
+		/* suspended state */
+		spi4_sleep: spi4_sleep {
+			drive-strength = <2>; /* 2 MA */
+			bias-pull-down; /* PULL Down */
+		};
+	};
+
+	spi4_cs0_active {
+		/* BLSP5 CS */
+		qcom,pins = <&gp 18>;
+		qcom,num-grp-pins = <1>;
+		qcom,pin-func = <1>;
+		label = "spi4-cs0-active";
+		spi4_cs0_active: cs0_active {
+			drive-strength = <2>;
+			bias-disable = <0>;
+		};
+	};
+
+	spi4_cs0_suspend {
+		/* BLSP5 CS */
+		qcom,pins = <&gp 18>;
+		qcom,num-grp-pins = <1>;
+		qcom,pin-func = <0>;
+		label = "spi4-cs0-suspend";
+		spi4_cs0_sleep: cs0_sleep {
+			drive-strength = <2>;
+			bias-disable = <0>;
+		};
+	};
 };
 
 &soc {
+	/delete-node/ spi@78B5000;
+	/delete-node/ i2c@7af5000;
+
+	spi_4: spi@7af5000 { /* BLSP2 QUP1 */
+		compatible = "qcom,spi-qup-v2";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg-names = "spi_physical", "spi_bam_physical";
+		reg = <0x7af5000 0x600>,
+		      <0x7ac4000 0x1f000>;
+		interrupt-names = "spi_irq", "spi_bam_irq";
+		/* IRQ 239 for BLSP2 */
+		interrupts = <0 299 0>, <0 239 0>;
+		spi-max-frequency = <19200000>;
+		pinctrl-names = "spi_default", "spi_sleep";
+		pinctrl-0 = <&spi4_default &spi4_cs0_active>;
+		pinctrl-1 = <&spi4_sleep &spi4_cs0_sleep>;
+		clock-names = "iface_clk", "core_clk";
+		clocks = <&clock_gcc clk_gcc_blsp2_ahb_clk>,
+			 <&clock_gcc clk_gcc_blsp2_qup1_spi_apps_clk>;
+		qcom,infinite-mode = <0>;
+		qcom,use-bam;
+		qcom,use-pinctrl;
+		qcom,ver-reg-exists;
+		/* need to check pipe index setting*/
+		qcom,bam-consumer-pipe-index = <4>;
+		qcom,bam-producer-pipe-index = <5>;
+		/* master-id for BLSP2 is MSM_BUS_MASTER_BLSP_2 or 84 */
+		qcom,master-id = <84>;
+		status = "ok";
+	};
+
 	/delete-node/ eldo2;
 
 	gpio_keys {
-- 
2.11.0

